game.cpp

```diff
@@ -548,6 +560,30 @@ player game::wall_collision(player p)
   return p;
 }
 
+void game::make_players_eat_food()
+{
+  int n_food = static_cast<int>(get_food().size());
+  for(auto& player : m_player)
+    {
+      for(int i = 0; i < n_food; ++i)
+       {
+          if (in_contact_with_uneaten_food(player, get_food()[i]))
+            {
+              eat_nth_food(i);
+              player.grow();
+            }
+       }
+    }
+}
+
+void game::eat_food(food& f)
```

Comment:
I am not sure, but I feel there is some redundancy here between this function and 
```c++
game::eat_nth_food()
```
As C++ core guidelines suggest, we should keep member functions to a minimum and instead use as many free functions as possible. So I would suggest to make 
```c++
game::eat_nth_food()
```
a free function again
```c++
eat_nth_food(game& g, const int n)
```
that calls in turn the member function
```c++
game::eat_food(food& f) 
```
@richelbilderbeek what do you think? Is this line of reasoning correct or we can allowe both functions to exist as member functions?

Reply:
@swom free functions, when possible, are to be preferred. 

Note that `eat_food` probably finds the index of the food item, so it can call `eat_nth_food`  :-)

Reply:
I don't like this situation either, and I did try to use `game::eat_nth_food()`, with no success. Can't recall why, I'll have a look again

Reply:
One day, there will be a member function `game::eat_food(std::iterator<food> iter)` (maybe called `remove_food` after a while), as using iterators is the preferred way to indicate an item in a vector. 

However, the team may not be ready yet to use iterators yet. But maybe there is a volunteer that wants to get more experience with algorithms and iterators.

For now, I feel using `game::eat_food`/`game::eat_nth_food` is a fine intermediary state.

Reply:
Ok, as per 685f64f `eat_nth_food()` is a free function again and `game::eat_food()` is now a public function (so `eat_nth_food()` can call it) 😁